Chapter_Version_Control.md

```diff
@@ -1,552 +1,619 @@
+
+Using branches keeps working code safe, particularly in collaborations. Each contibuter can have their own branch or branches which are only merged into the main project when they are ready.
+
+![sub_branch](figures/sub_branch.png)
 
-You need to resolve the conflict manually. Make changes to lib/hello.html to achieve the following result.
 
-FILE: LIB/HELLO.HTML
+### How to do it
+
+You can create a branch and switch to it using:
+```
+git checkout -b name_of_your_new_branch
+```
+
+To change between branches:
 ```
-<!-- Author: Alexander Shvets (alex@githowto.com) -->
-<html>
-  <head>
-    <link type="text/css" rel="stylesheet" media="all" href="style.css" />
-  </head>
-  <body>
-    <h1>Hello, World! Life is great!</h1>
-  </body>
-</html>
+$ git checkout name_of_the_branch
 ```
-03 Make a commit of conflict resolution
 
-RUN:
+though you must commit any work you have in progress before you will be able to switch. You can see all branches of your project simply using
 ```
-git add lib/hello.html
-git commit -m "Merged master fixed conflict."
+$ git branch
 ```
+which will output a list with an asterix next to the branch you're on. You can also use `git status` if you've forgotten which branch you're on.
 
-RESULT:
+If you decide to get rid of a branch you can delete it with
 ```
-$ git add lib/hello.html
-$ git commit -m "Merged master fixed conflict."
-Recorded resolution for 'lib/hello.html'.
-[style 645c4e6] Merged master fixed conflict.
+$ git branch -D name_of_the_branch
 ```
 
-------
+### Good practice for branches
 
-From [here](http://genomewiki.ucsc.edu/index.php/Resolving_merge_conflicts_in_Git) **["You are granted a limited license to copy anything from this site"](http://genomewiki.ucsc.edu/index.php/Genomewiki:General_disclaimer)**
+Branches should be used to **keep the master branch clean**. Similarly you should try to keep individual branches as clean as possible by **only adding one new feature per branch**, because if you are working on several features some may be finished and ready to merge into master while others are stull under development. Give your branches **sensible names**, "new_feature" is all well and good until you start developing a newer feature on another branch.
```

Comment:
Definition looks good, thanks for adding!

I would still like to get something on branching models in the book somewhere (even if only a single mention of GitHub flow), mainly to discourage getting into a messy state due to too much branching off branches.

Shall we icebox the idea for now and come back to it later? It might fit better into a discussion of versioning and releases, or collaboration in one of the other chapters.

Reply:
Yep, sounds good.